titus-server-master/src/main/java/com/netflix/titus/master/scheduler/fitness/AgentManagementFitnessCalculator.java

```diff
@@ -88,6 +95,22 @@ public double calculateFitness(TaskRequest taskRequest, VirtualMachineCurrentSta
             }
         }
 
-        return NOT_ACTIVE_INSTANCE_GROUP_SCORE;
+        String instanceId = getAttributeValueOrEmptyString(targetVM, schedulerConfiguration.getInstanceAttributeName());
```

Comment:
This section should be in the `if` statement above.

Reply:
Yes, this code will never execute. I am going to remove this for now as we need to figure out what the behavior should be.